xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/web/XWikiServletResponse.java

```diff
@@ -66,9 +68,21 @@ public void sendRedirect(String redirect) throws IOException
             LOGGER.warn("Possible HTTP Response Splitting attack, attempting to redirect to [{}]", redirect);
             return;
         }
+        if (redirect.matches("[a-z0-9]+://.*")) {
```

Comment:
Shouldn't we display something if we don't redirect? At least some warning log?

Reply:
Not sure to understand what you suggest here: we do output a warning log if we don't redirect. Here this if is just to know if we need to check the domain.

Reply:
ok I was just checking the diff and saw no ELSE. I've checked the file now and it seems that if it doesn't match we redirect without a check. So my understanding now (BTW a comment would be nice just before the match() call to explain what it's doing and why it's there) is that we check only when there's a URL scheme and the reason is probably because we don't want to perform the check for relative URLs since they're on the same domain (BTW I had to check `sendRedirect` again and indeed it supports relative URLs: `This method can accept relative URLs;the servlet container must convert the relative URL to an absolute URL before sending the response to the client. If the location is relative without a leading '/' the container interprets it as relative to the current request URI. If the location is relative with a leading '/' the container interprets it as relative to the servlet container root.`).

So it looks good, except the missing comment. 

Thanks